### PR TITLE
ipc4-topology: fixup process module support

### PR DESCRIPTION
**Pull request by aiChaoSONG**
move the token used in process module support from struct snd_sof_widget to to process specific struct.

### CODE REVIEW

**Review by bardliao — DISMISSED** [review]

File: include/uapi/sound/sof/tokens.h [hunk 1]
```diff
@@ -133,6 +132,7 @@
 
 /* Processing Components */
 #define SOF_TKN_PROCESS_TYPE                    900
+#define SOF_TKN_PROCESS_INIT_BLOB_HAS_OUT_FMT   901
```
Comment (ujfalusi) [inline include/uapi/sound/sof/tokens.h]: Any hint on the why?

File: sound/soc/sof/ipc4-topology.h [hunk 2]
```diff
@@ -349,6 +349,12 @@ struct sof_ipc4_process {
 	void *ipc_config_data;
 	uint32_t ipc_config_size;
 	struct sof_ipc4_msg msg;
+	/*
+	 * init_blob_has_out_fmt is used to describe whether there is output format
+	 * (struct sof_ipc4_audio_format output_format) in the module init instance
+	 * ipc message payload blob.
+	 */
+	bool init_blob_has_out_fmt;
```
Comment (ujfalusi) [inline sound/soc/sof/ipc4-topology.h]: this flag only applies to the init message? Is there other message regarding to the module which is not init?

**Review by ujfalusi — COMMENTED** [review]
@aiChaoSONG, this is quite a big fixup, completely making the initial process support incompatible with what we have after this fixup, can you explain why is the rework?

File: include/uapi/sound/sof/tokens.h (hunk 1 above)
Comment (aiChaoSONG) [reply include/uapi/sound/sof/tokens.h]: I add comment in the variable in snd_sof_process struct, I think we don't need comments at both places, right?

File: sound/soc/sof/ipc4-topology.h (hunk 2 above)
Comment (aiChaoSONG) [reply sound/soc/sof/ipc4-topology.h]: yes, this flag is only for init message. there are other messages for the module(set_config, set_large_config), but none of them touch this flag.

File: include/uapi/sound/sof/tokens.h (hunk 1 above)
Comment (ujfalusi) [reply include/uapi/sound/sof/tokens.h]: I meant for this change...
Originally you used completely different token.

Comment (ujfalusi) [reply include/uapi/sound/sof/tokens.h]: and you obviously need to provide a new commit message for the original commit as the `SOF_TKN_COMP_PAYLOAD_WITH_OUTPUT_FMT` is gone with this "fixup"

Comment (aiChaoSONG) [reply include/uapi/sound/sof/tokens.h]: if this flag is true, it means there is output format in the init blob, we need to append the format to the init blob (after base module config).

module init blob construct steps:
mandatory base module config + [optional output format] + [optional module specific extension if given from topology]

File: sound/soc/sof/ipc4-topology.h (hunk 2 above)
Comment (ujfalusi) [reply sound/soc/sof/ipc4-topology.h]: Right, and if the blob have "source format"? ;)
What was wrong with the original "payload_with_output_fmt" name of the flag?

File: include/uapi/sound/sof/tokens.h (hunk 1 above)
Comment (ranj063) [reply include/uapi/sound/sof/tokens.h]: > I meant for this change... Originally you used completely different token.

@ujfalusi I think this is OK. luckily this token isnt currently used in any of our topologies. The change is needed becaue we're moving away from being a COMP_TOKEN to a PROCESS_TKN

File: sound/soc/sof/ipc4-topology.h (hunk 2 above)
Comment (ranj063) [reply sound/soc/sof/ipc4-topology.h]: I agree we should keep the same flag name too.

Comment (aiChaoSONG) [reply sound/soc/sof/ipc4-topology.h]: > Right, and if the blob have "source format"? ;) What was wrong with the original "payload_with_output_fmt" name of the flag?

Nothing wrong here, I just want to be more precise. even `payload_with_output_fmt`, it really means the init_payload_with_output_fmt, just as I said, this token is only used with the module init ipc message payload, not elsewhere.

@ranj063 @ujfalusi If you insist, I can keep the name `payload_with_output_fmt`,  

btw, in term of the token name SOF_TKN_PROCESS_PAYLOAD_WITH_OUTPUT_FMT, I think it is too long, it has 39 letters.

File: include/uapi/sound/sof/tokens.h (hunk 1 above)
Comment (aiChaoSONG) [reply include/uapi/sound/sof/tokens.h]: > and you obviously need to provide a new commit message for the original commit as the `SOF_TKN_COMP_PAYLOAD_WITH_OUTPUT_FMT` is gone with this "fixup"

I will add some explanation to why I do a rename, and why we move it from a comp token to process module token.

File: sound/soc/sof/ipc4-topology.h (hunk 2 above)
Comment (ranj063) [reply sound/soc/sof/ipc4-topology.h]: please shorten the token name as needed. As long as it matches the description, it should be fine

**Comment by aiChaoSONG** [comment]
@ujfalusi @ranj063 Now the fixup looks more smooth. basic idea is keep process module related tokens or variables process module specific

**Review by ranj063 — DISMISSED** [review]

**Comment by aiChaoSONG** [comment]
new update: add SOF_PROCESS_TOKENS to process_token_list

**Review by ranj063 — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

File: include/uapi/sound/sof/tokens.h [hunk 3]
```diff
@@ -133,6 +132,7 @@
 
 /* Processing Components */
 #define SOF_TKN_PROCESS_TYPE                    900
+#define SOF_TKN_PROCESS_PAYLOAD_WITH_OUTPUT_FMT 901
```
Comment (ujfalusi) [inline include/uapi/sound/sof/tokens.h:135]: Let's hope that we will not forget to update the commit subject and message when squashing the fixup for upstream...

**Review by ujfalusi — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by ranj063** [comment]
SOFCI TEST

**Pull request merged**